[19.0][MIG] database_cleanup: Migration to 19.0

bizzappdev opened the pull request.

Commit:
[ADD] Database cleanup module

Commit:
remove relations when purging models
avoid ''NoneType' object has no attribute 'exists'' error when purging models
fix my change guewen.baconnier@camptocamp.com-20140203103254-v1mzu2uib047xb9h, wrong lines replaced...

Commit:
[FIX] Don't remove uid field from wkf_instance, which is written in
    raw SQL query (but never read afterwards). Workaround for
    lp:1277899

[FIX] Preserve dangling workflow table which is in use

[RFR] Group models per table when detecting columns to purge
      to prevent problems with models sharing the same table

[ADD] Allow purging of dangling data entries

[FIX] Data purging now working

[IMP] Docstrings

[FIX] Label
[FIX] Catch attempt to unlink field from nonexisting model

[RFR] Flake8

anthony-muschang committed:
[CHG] Migration to 8.0

[CHG] database_cleanup: move description to README.rst

hbrunn committed:
[IMP] hide unnecessary buttons in wizard

[IMP] order wizard lines by name

[IMP] deal with modules whose models can't be loaded

[IMP] double quotes for docstring

[FIX] use exists query instead of huge in list

[IMP] hide unnecessary buttons in wizard II

[IMP] readability

[FIX] cope with purging nonexisting models

hbrunn committed:
[IMP] allow to select lines to purge in a tree view

[IMP] adaptto upstream changes

sysadminmatmoz committed:
Missing templates and translations added

hbrunn committed:
[ADD] allow to clean up menus

hbrunn committed:
[ADD] migrate database_cleanup

[ADD] test purging modules

[ADD] test purging tables

hbrunn committed:
[FIX] purge uninstalled uninstallable modules instead of deleting

[FIX] database_cleanup reloads the registry

which has weird side effects during testing. Take care
database_cleanup's tests don't mess up the following tests

StefanRijnhart committed:
[RFR] Explicit access rights so that tests can run
Fixes #505

hbrunn committed:
[IMP] call the button handler to also purge reverse dependencies

hbrunn committed:
[FIX] don't crash if an xmlid refers to a nonexisting field (#559)

JordiBForgeFlow committed:
[FIX] clean database menus was proposing good menus to purge (#562)

florian-dacosta committed:
fixes #587

moylop260 committed:
[FIX] database_cleanup: Fix test (#612)

lasley committed:
[9.0][FIX] database_cleanup: Isolate build (#719)

* [FIX] database_cleanup: Isolate build
* Isolate `database_cleanup` into its own build in Travis file to fix #689

* [FIX] database_cleanup: Remove KeyError assertion
* Remove KeyError assertion in tests due to PR in comment being merged

hbrunn committed:
[ADD] allow creating missing indexes and purging properties (#736)

* [ADD] allow creating missing indexes

* [FIX] tests; installation

* [ADD] allow purging properties

* [ADD] missing file

* [ADD] test purging properties

* [ADD] missing parent_id for menu entry

* [FIX] don't delete too many and wrong properties

eantones committed:
[MIG] database_cleanup: Migration to version 10.0

Using new base model inheritance.

hbrunn committed:
[FIX] really uninstall modules and avoid a crash on cached data

[FIX] don't try to uninstall uninstalled modules

[DEL] weird code

[FIX] actually cleanup where we can

hbrunn committed:
[ADD] [database_cleanup] migrate to 11.0

fanha99 committed:
database_cleanup_fix_runbot_links

simahawk committed:
Global pylint cleanup

mart-e committed:
[MIG] database_cleanup: migration to 12.0

Update version number
Use correct import
Update .pot to match new syntax
Adapt icon to fontawesome
Add models description
Rename field external id to new syntax <model>__<field>

hbrunn committed:
[FIX] don't destroy values when there are empty default properties

hbrunn committed:
[FIX] show the redundant property's res_id, not the default one's

hbrunn committed:
[IMP] don't break on properties pointing to unknown models

mart-e committed:
[FIX] database_cleanup: exclude password fields

cf discussion on OCA/server-tools#1408

mart-e committed:
[I18N] database_cleanup: reflect changes to en.po

To match the new database_cleanup.pot

mart-e committed:
[FIX] database_cleanup: filter on many2many fields

Following odoo/odoo@54238c88e85d6e a Many2many may have no relation
This avoids having a None record in the list (which is problematic with the
NOT IN clause)
https://wiki.postgresql.org/wiki/Don't_Do_This#Don.27t_use_NOT_IN

mart-e committed:
[FIX] database_cleanup: ignore to_buy modules

Since 12.0, some virtual modules are created as an incentive to buy enterprise
Hide installation warning

bizzappdev committed:
[FIX][#1580]database_cleanup: Skip studio_customization module skipped for uninstalling

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (74 of 74 strings)

Translation: server-tools-12.0/server-tools-12.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-database_cleanup/pt_BR/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 51.4% (38 of 74 strings)

Translation: server-tools-12.0/server-tools-12.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-database_cleanup/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (74 of 74 strings)

Translation: server-tools-12.0/server-tools-12.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-12-0/server-tools-12-0-database_cleanup/zh_CN/

Commit:
[IMP] database_cleanup: black, isort, prettier

Commit:
[MIG] database_cleanup: Migration to 14.0

legalsylvain committed:
[IMP] database_cleanup : delete also obsolete SQL views

Co-authored-by: Pedro M. Baeza <pedro.baeza@tecnativa.com>

Commit:
fixup! [MIG] database_cleanup: Migration to 14.0

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (74 of 74 strings)

Translation: server-tools-14.0/server-tools-14.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-database_cleanup/es_AR/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 20.2% (15 of 74 strings)

Translation: server-tools-14.0/server-tools-14.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-14-0/server-tools-14-0-database_cleanup/it/

JoelZilli committed:
[MIG] database_cleanup: Migration to 16.0

ypapouin committed:
[ADD] database_cleanup: Purge fields (ir.model.fields) added

yankinmax committed:
[IMP] database_cleanup: improve tests

JoelZilli committed:
[FIX] database_cleanup: unused code + wrong modules clean

StefanRijnhart committed:
[FIX] database_cleanup: preserve spaces in quoted identifiers

miikanissi committed:
[IMP] database_cleanup: Readability of modules to remove

miikanissi committed:
[FIX] database_cleanup: TransactionCase already does rollback

No need to rollback here as TransactionCase does a safe rollback.

miikanissi committed:
[IMP] database_cleanup: Unlink with new cursor

Needed for purging payment providers which recompute module_state

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-16.0/server-tools-16.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-database_cleanup/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (76 of 76 strings)

Translation: server-tools-16.0/server-tools-16.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-database_cleanup/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.6% (75 of 76 strings)

Translation: server-tools-16.0/server-tools-16.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-database_cleanup/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (76 of 76 strings)

Translation: server-tools-16.0/server-tools-16.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-16-0/server-tools-16-0-database_cleanup/it/

augusto-weiss committed:
[IMP] database_cleanup: pre-commit execution

augusto-weiss committed:
[MIG] database_cleanup: Migration to 17.0

Commit:
[FIX] database_cleanup: fix logging error introduced by #2098

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-17.0/server-tools-17.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-17-0/server-tools-17-0-database_cleanup/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-17.0/server-tools-17.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-17-0/server-tools-17-0-database_cleanup/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (75 of 75 strings)

Translation: server-tools-17.0/server-tools-17.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-17-0/server-tools-17-0-database_cleanup/it/

ypapouin committed:
[FIX] database_cleanup: Check that the model is automatically managed by Odoo

MrGigSolutions committed:
Prevent deletion of columns added in the init method of the model.

See issue https://github.com/OCA/server-tools/issues/2851.

astirpe committed:
[MIG] add tests for purge fields

pedrobaeza committed:
[FIX] *: Pre-commit fixes

astirpe committed:
[MIG] adapt code for V17

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-17.0/server-tools-17.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-17-0/server-tools-17-0-database_cleanup/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (81 of 81 strings)

Translation: server-tools-17.0/server-tools-17.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-17-0/server-tools-17-0-database_cleanup/it/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-17.0/server-tools-17.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-17-0/server-tools-17-0-database_cleanup/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (84 of 84 strings)

Translation: server-tools-17.0/server-tools-17.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-17-0/server-tools-17-0-database_cleanup/it/

xtanuiha committed:
Translated using Weblate (Chinese (Simplified) (zh_CN))

Currently translated at 100.0% (84 of 84 strings)

Translation: server-tools-17.0/server-tools-17.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-17-0/server-tools-17-0-database_cleanup/zh_CN/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (84 of 84 strings)

Translation: server-tools-17.0/server-tools-17.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-17-0/server-tools-17-0-database_cleanup/es_AR/

nguyenminhchien committed:
[MIG] database_cleanup: Migration to 18.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-18.0/server-tools-18.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-18-0/server-tools-18-0-database_cleanup/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-18.0/server-tools-18.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-18-0/server-tools-18-0-database_cleanup/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (75 of 75 strings)

Translation: server-tools-18.0/server-tools-18.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-18-0/server-tools-18-0-database_cleanup/it/

StefanRijnhart committed:
[IMP] database_cleanup: add table purge blacklist with known non-model table

StefanRijnhart committed:
[IMP] database_cleanup: set appropriate labels in each purge view

instead of 'Purge this model' everywhere.

TomTietze committed:
Translated using Weblate (German)

Currently translated at 100.0% (75 of 75 strings)

Translation: server-tools-18.0/server-tools-18.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-18-0/server-tools-18-0-database_cleanup/de/

nicomacr commented:
Hi @bizzappdev,
I've started the migration on this [PR](https://github.com/OCA/server-tools/pull/3454)
I need this module to be installable and this one is not able to install yet.

bizzappdev commented:
We are working on the migration. And almost everything is migrated. 
QA guys are testing. Once it is ready within 1 or 2 days, we will commit the migrated version.
@nicomacr

bizzappdev commented:
@nicomacr modules is migrated and ready for the review.

nicomacr reviewed (approved):
LGTM

twalter-c2c commented on database_cleanup/tests/common.py:
```diff
@@ -17,7 +17,9 @@ def environment():
+        yield env
 
 
 # Use post_install to get all models loaded more info: odoo/odoo#13458
```
Is there a reason not to replace `setUp()` with `setUpClass()` at lines 28-29?

twalter-c2c commented on database_cleanup/tests/test_identifier_adapter.py:
```diff
@@ -1,9 +1,8 @@
 
 
-class TestIdentifierAdapter(TransactionCase):
+class TestIdentifierAdapter(BaseCommon):
```
Is there a need to use the `BaseCommon` here? Maybe you can make use of disabling tracking: `cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))` as advised [here](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-19.0#:~:text=Disable,way%29%2E) instead? I believe that disabled tracking is recommended for all tests.

twalter-c2c reviewed (commented):
Looks good in general. I am about to test the module functionally. I noticed that you did some refactoring - you moved code blocks for each model to a dedicated file, thank you for that. However, I would really appreciate it if you could keep the refactoring in a separate commit to ease reviewing the migration changes.

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: server-tools-18.0/server-tools-18.0-database_cleanup
Translate-URL: https://translation.odoo-community.org/projects/server-tools-18-0/server-tools-18-0-database_cleanup/

twalter-c2c commented:
It would be great if you could rebase on top of this PR https://github.com/OCA/server-tools/pull/3273 since it got merged.

bizzappdev committed:
[IMP] database_cleanup: pre-commit auto fixes

bizzappdev committed:
[MIG] database_cleanup: Migration to 19.0

bizzappdev committed:
[IMP] database_cleanup: Improved the file structure

twalter-c2c reviewed (approved):
Thank you for the updates. It looks good now.

simahawk commented:
/ocabot merge nobump
